frontend/src/plugin/Plugins.tsx

```diff
@@ -0,0 +1,46 @@
+import { useEffect } from 'react';
+import { useDispatch } from 'react-redux';
+import helpers from '../helpers';
+import { setPluginsLoadState } from '../redux/actions/actions';
+import { initializePlugins } from './index';
+
+/**
+ * For discovering and executing plugins.
+ */
+export default function Plugins() {
+  const dispatch = useDispatch();
+
+  // only run on first load
+  useEffect(() => {
+    const fetchData = async () => {
+      const pluginsScriptPaths = (await fetch(`${helpers.getAppUrl()}plugins/list`).then(resp =>
+        resp.json()
+      )) as string[];
+
+      const sources = await Promise.all(
+        pluginsScriptPaths.map(path =>
+          fetch(`${helpers.getAppUrl()}${path}`).then(resp => resp.text())
+        )
+      );
+
+      sources.forEach(source => {
+        (function (str: string) {
+          try {
+            const result = eval(str);
+            return result;
+          } catch (e) {
+            console.error(e);
+          }
+        }.call({}, source));
```

Comment:
I'm surprise this works fine like this. Before we were injecting the scripts into the document. Not sure how things differ or what's best, just curious.